### PR TITLE
Update .gitignore templates

### PR DESCRIPTION
**Pull request by fertapric**
The new templates are similar to the ones generated by `mix new`, which contains more detailed descriptions and more files/directories ignored. I noticed it after running `mix test --cover` and seeing that the `cover` directory was not included by default.

Besides, it appends slashes to the name of directories to differentiate them from regular files. This change [was discussed on the Elixir core mailing list](https://groups.google.com/forum/#!topic/elixir-lang-core/1uf4pC5Iyvw), and [later implemented on the .gitignore templates of `mix new`](https://github.com/elixir-lang/elixir/pull/6091).

In addition, it includes in the list of ignored files the package tarballs generated with `mix hex.build`. These files are also included in the .gitignore templates of `mix new`. This change was introduced in elixir-lang/elixir#6494.

This PR also changes the current generators to use specific .gitignore templates for umbrella apps. The gitignore template located at the path [`installer/templates/phx_umbrella/apps/app_name_web/gitignore`](https://github.com/phoenixframework/phoenix/blob/master/installer/templates/phx_umbrella/apps/app_name_web/gitignore) was not being used. This commit uses that template and includes a custom template for the non-web app inside the umbrella.

### CODE REVIEW

File: installer/lib/phx_new/web.ex [hunk 1]
```diff
@@ -58,12 +58,9 @@ defmodule Phx.New.Web do
     {:eex,  "phx_test/views/page_view_test.exs",              :web, "test/:web_app/views/page_view_test.exs"},
   ]
 
-  template :bare, [
-    {:text, "phx_assets/bare/gitignore", :web, ".gitignore"},
-  ]
+  template :bare, []
```
Comment (fertapric) [inline installer/lib/phx_new/web.ex]: I was not sure if leaving or removing this line.

Other option is to mimic the directory structure for single apps and create `brunch` and `bare` directories with two `gitignore` in them.

File: installer/lib/phx_new/single.ex [hunk 2]
```diff
@@ -42,7 +42,7 @@ defmodule Phx.New.Single do
   ]
 
   template :brunch, [
-    {:text, "phx_assets/brunch/gitignore",        :project, ".gitignore"},
+    {:eex,  "phx_assets/brunch/gitignore",        :project, ".gitignore"},
```
Comment (fertapric) [inline installer/lib/phx_new/single.ex]: I kept the current directory structure of `brunch` and `bare`, but I think we could merge both `gitignore` templates inside `phx_single/gitignore` and have an `if brunch do` to ignore `npm-debug.log` and `/node_modules/`.

Wdyt?

File: installer/templates/phx_assets/bare/gitignore [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
+# If the VM crashes, it generates a dump, let's ignore it too.
 erl_crash.dump
 
+# Also ignore archive artifacts (built via "mix archive.build").
+*.ez
+
+# Ignore package tarball (built via "mix hex.build").
+<%= app_name %>-*.tar
+
+# Database data
```
Comment (OvermindDL1) [inline installer/templates/phx_assets/bare/gitignore]: What database is this for?  I know it's a common Mnesia directory, but this doesn't say that?

Comment (fertapric) [reply installer/templates/phx_assets/bare/gitignore]: To be honest, I saw it in the previous template but I was not sure in which scenarios that directory was generated.

Glad to help the description to something better :) Thanks!

File: installer/templates/phx_umbrella/apps/app_name_web/gitignore [hunk 4]
```diff
@@ -1,14 +1,35 @@
...
+*.ez
+
+# Ignore package tarball (built via "mix hex.build").
+<%= web_app_name %>-*.tar
+<%= if brunch do %>
+# If NPM crashes, it generates a log, let's ignore it too.
+npm-debug.log
+
+# The directory NPM downloads your dependencies sources to.
+/node_modules/
```
Comment (fertapric) [inline installer/templates/phx_umbrella/apps/app_name_web/gitignore]: This was on the original template, but it seems it should be `/assets/node_modules/`

File: installer/test/phx_new_test.exs [hunk 5]
```diff
@@ -74,7 +74,8 @@ defmodule Mix.Tasks.Phx.NewTest do
                   "<title>Hello PhxBlog!</title>"
 
       # Brunch
-      assert_file "phx_blog/.gitignore", "/node_modules"
+      assert_file "phx_blog/.gitignore", "/node_modules/"
```
Comment (fertapric) [inline installer/test/phx_new_test.exs]: Should I change those tests to match `/assets/node_modules/`? I would say so.

File: installer/templates/phx_assets/bare/gitignore (hunk 3 above)
Comment (josevalim) [reply installer/templates/phx_assets/bare/gitignore]: sqlite3

File: installer/templates/phx_umbrella/apps/app_name_web/gitignore (hunk 4 above)
Comment (josevalim) [reply installer/templates/phx_umbrella/apps/app_name_web/gitignore]: Yeah.

File: installer/test/phx_new_test.exs (hunk 5 above)
Comment (josevalim) [reply installer/test/phx_new_test.exs]: Yeah.

Comment (fertapric) [reply installer/test/phx_new_test.exs]: Done in https://github.com/phoenixframework/phoenix/pull/2772/commits/61f9544e8322bc03c809678d0dbfe0b3a5a62e39

File: installer/templates/phx_umbrella/apps/app_name_web/gitignore (hunk 4 above)
Comment (fertapric) [reply installer/templates/phx_umbrella/apps/app_name_web/gitignore]: Done in https://github.com/phoenixframework/phoenix/pull/2772/commits/61f9544e8322bc03c809678d0dbfe0b3a5a62e39

File: installer/templates/phx_assets/bare/gitignore (hunk 3 above)
Comment (fertapric) [reply installer/templates/phx_assets/bare/gitignore]: Changed to:

```
# Ignore the SQLite 3 database files.
/db/*.sqlite3
/db/*.sqlite3-journal
```

Done in https://github.com/phoenixframework/phoenix/pull/2772/commits/e111611753bbc47f955aca45632e1c08da71280e

Comment (fertapric) [reply installer/templates/phx_assets/bare/gitignore]: On the other hand, [the generator does not include `sqlite` in the list of Ecto adapters](https://github.com/phoenixframework/phoenix/blob/master/installer/lib/phx_new/generator.ex#L178-L189), should we then remove the `/db` directory from the template?

The line `/db` was introduced in the previous template [in this commit](https://github.com/phoenixframework/phoenix/commit/f0f8c5341f485ffbd1222dda06d02a4c552664ea). However, support for SQLite was removed in [this commit](https://github.com/phoenixframework/phoenix/commit/004ed0bd7618f297569a5abc0abc487ae215b1a2#diff-62c3c1595e6fe80fd313b78665b1f44d).

**Comment by josevalim** [comment]
Yeah, we can remove it.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

File: installer/templates/phx_assets/bare/gitignore (hunk 3 above)
Comment (fertapric) [reply installer/templates/phx_assets/bare/gitignore]: Removed in https://github.com/phoenixframework/phoenix/pull/2772/commits/01caf31d67f88bcda77470adf6e877f4a6856e0e

**Comment by fertapric** [comment]
Feedback applied. Test are failing for things that don't seem to be related to this PR.

File: installer/lib/phx_new/single.ex (hunk 2 above)
Comment (josevalim) [reply installer/lib/phx_new/single.ex]: Merging sounds good to me, since that's what we have done below anyway.

**Comment by josevalim** [comment]
@fertapric I have added one comment and we should be good to go!

**Comment by fertapric** [comment]
@josevalim I've addressed your comment in https://github.com/phoenixframework/phoenix/pull/2772/commits/0be5b158925a13430be7156aa33089253ac4db24 :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**